webapp/src/shared/components/templates/RowsPageTemplate/RowsPageTemplate.tsx

```diff
@@ -0,0 +1,82 @@
+import './RowsPageTemplate.css';
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { USER_URL, AVATAR_EP_URL, WILDCARD_AVATAR_URL } from '../../../urls';
+import { MediumAvatar } from '../../Avatar/Avatar';
+import SearchForm from '../../Input/SearchForm';
+import RowsList, { RowItem } from '../../RowsList/RowsList';
+import NavigationBar from '../../NavigationBar/NavigationBar';
+import { useData } from '../../../hooks/UseData';
+import Loading from '../../Loading/Loading';
+import { usersApi } from '../../../services/ApiService';
+import { useSearchContext } from '../../../context/SearchContext';
+
+type RowsPageTemplateProps<T> = {
+  dataValidator: (data: T) => boolean;
```

Comment:
The data validator is there to look at whether the type of the object is T. I don't think it makes sense to have the data validator take T as an argument. It's a runtime check.